src/pages/community/events/high-performance-tidb-challenge.js

```diff
@@ -0,0 +1,577 @@
+import React, { useState, useEffect } from 'react'
+import Link from '../../../components/IntlLink'
+import Layout from '../../../components/layout'
+import SEO from '../../../components/seo'
+import {
+  noticeItemData,
+  noticeDetailsData,
+  materialsData,
+} from '../../../data/community/events/hptc'
+import axios from 'axios'
+
+import { trackCustomEvent } from 'gatsby-plugin-google-analytics'
+
+const HPTC = ({ data }) => {
+  const {
+    heroBg,
+    taskCard,
+    taskCardMobile,
+    step1Bg,
+    step4Bg,
+    usabilityCard,
+    performanceCard,
+  } = data
+
+  // track sign up button click event
+  const trackViews = (event, category) => {
+    trackCustomEvent({
+      category: category,
+      action: 'click',
+      label: event,
+    })
+  }
+
+  const [rankList, setRankList] = useState([])
+  const [signUpBannerLink, setSignUpBannerLink] = useState('')
+  const [signUpStepLink, setSignUpStepLink] = useState('')
+  const http = axios.create({
+    baseURL: 'https://bots.tidb.io/ti-challenge-bot/program/2/ranks',
+  })
+
+  // append params to track media source
+  const getURLParams = () => {
+    let utmSource = 'pingcap'
+    let utmMediuBanner = 'banner'
+    let utmMediumStep = 'step'
+    let utmCampaign = 'hptc'
+    let params = window.location.search.slice(1).split('&')
+
+    let paramsObj = {}
+    if (params.length > 1) {
+      params.forEach((p) => {
+        paramsObj[p.split('=')[0]] = p.split('=')[1]
+      })
+
+      utmSource = paramsObj['utm_source']
+      utmMediuBanner = paramsObj['utm_medium']
+      utmMediumStep = paramsObj['utm_medium']
+      utmCampaign = paramsObj['utm_campaign']
+    }
+
+    let registerBtnURLBanner =
+      'https://forms.pingcap.com/f/high-performance-challenge-en?' +
+      'utm_source=' +
+      utmSource +
+      '&utm_medium=' +
+      utmMediuBanner +
+      '&utm_campaign=' +
+      utmCampaign
+
+    let registerBtnURLStep =
+      'https://forms.pingcap.com/f/high-performance-challenge-en?' +
+      'utm_source=' +
+      utmSource +
+      '&utm_medium=' +
+      utmMediumStep +
+      '&utm_campaign=' +
+      utmCampaign
+
+    setSignUpBannerLink(registerBtnURLBanner)
+    setSignUpStepLink(registerBtnURLStep)
+  }
+
+  useEffect(() => {
+    getURLParams()
+  }, [signUpBannerLink, signUpStepLink])
+
+  useEffect(() => {
+    async function getRankList() {
+      try {
+        const rankListData = (await http.get()).data.data
+        setRankList(rankListData)
+      } catch (e) {
+        console.log('Get Rank List Failed', e)
+        return
+      }
+    }
+
+    getRankList()
+  }, [])
+
+  const openNoticeDetail = (idx) => {
+    let className = 'show-detail'
+    if (window.matchMedia('(max-width:550px)').matches) {
+      className = 'detail-mobile'
+    }
+
+    const allEle = document.getElementsByClassName(className)
+    Array.from(allEle).forEach((ele) => {
+      ele.classList.remove(className)
+    })
+
```

Comment:
In React, it is inadvisable to mutate DOM element directly using DOM API, since all changes you did in dom will be overided  by React reconciliation after rerender. Maybe it is ok for now, but still unstable. You can use state to save className or use ref to save invariable dom element during multiple render.

Reply:
You're right. It's has been updated to use ref to access DOM.